### PR TITLE
Proxy client requests through the python app. Store access tokens encrypted in cookies.

### PR DESCRIPTION
**Pull request by mfjarvis**

### CODE REVIEW

**Comment by mfjarvis** [comment]
The release pipeline is currently set to deploy this branch to Heroku. Once merged, change back to master.

File: client/app.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            "redirect_uri": REDIRECT_URI,
+            "client_id": CLIENT_ID,
+            "client_secret": CLIENT_SECRET,
+        }
+    )
+    return token_response.json()
+
+
+if __name__ == "__main__":
+    app.run(port=5000)
```
Comment (ri-ch) [inline client/app.py:124]: For dev purposes we probably want debug = True

File: client/app.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    code = flask.request.args.get("code")
+    state = flask.request.args.get("state")
+
+    token_response_json = exchange_code_for_token(code)
+    access_token = token_response_json["access_token"]
+    expires_in = token_response_json["expires_in"]
+    access_token_encrypted = fernet.encrypt(access_token.encode('utf-8')).decode('utf-8')
+    expires = datetime.datetime.utcnow() + datetime.timedelta(seconds=float(expires_in))
+
+    callback_response = flask.redirect(f"/{state}", 302)
```
[the remark this answers is not in the document]
Comment (mfjarvis) [reply client/app.py]: Removed 302s, added map from state to redirect URL so we only redirect to allowed URLs.

File: client/app.py (hunk 1 above)
Comment (mfjarvis) [reply client/app.py]: Done

**Pull request merged**